src/user-status/userStatusReducers.js

```diff
@@ -0,0 +1,41 @@
+/* @flow strict-local */
+import type { UserStatusState, Action } from '../types';
+import {
+  LOGOUT,
+  LOGIN_SUCCESS,
+  ACCOUNT_SWITCH,
+  REALM_INIT,
+  EVENT_USER_STATUS_UPDATE,
+} from '../actionConstants';
+import { NULL_OBJECT } from '../nullObjects';
+
+const initialState: UserStatusState = NULL_OBJECT;
```

Comment:
Completely disagree because:
 * we are currently using this pattern everywhere
 * there are comparisons happening, any selector that uses that or any component that uses that as props does a comparison against the old value
 * as a bonus this object is frozen and will throw if we mutate it, revealing a potential bug

Reply:
> there are comparisons happening, any selector that uses that or any component that uses that as props does a comparison against the old value

Right, but that's perfectly well served by `const initialState: UserStatusState = {};`. 

(That's why I wrote "with some *other* unrelated object that might happen to be empty".)

If there's ever a place where we compare one of these with a value that comes from some *other* mention of `NULL_OBJECT` -- like some other reducer for a different part of the state tree -- that would indicate a very worrying bug.


> we are currently using this pattern everywhere

Hmm, true:
```
$ perl -lne 'print $1 if (/^const initialState.* = (.*)/)' src/*/*Reducer*.js | sort | uniq -c
      5 {
      1 getStateForRoute('loading') || NULL_NAV_STATE;
     12 NULL_ARRAY;
      9 NULL_OBJECT;
```

So *shrug* I guess it's fine for this PR to go along with the pattern, then. I might go through at some point and see about changing them all together, though.

Reply:
> as a bonus this object is frozen and will throw if we mutate it, revealing a potential bug

This is certainly something that could be useful in general!

I am not sure there is much value, though, in only having it on the empty values. I'd rate it a lot more likely to catch bugs if we were to do it systematically, on all the objects and arrays in our Redux state trees. The Reselect cached selector values would also be a good target.

Reply:
I suspect the operation on deep object is slow as `Object.freeze()` does not work deeply and the `deepFreeze` we use in the unit-tests is iterating children calling `Object.freeze()` on them if array or object.

Reply:
It seems the source code for that is shorter than an explanation:
https://github.com/substack/deep-freeze/blob/master/index.js

Reply:
> It seems the source code for that is shorter than an explanation

Like so much of the JS library ecosystem. :stuck_out_tongue_closed_eyes: Thanks for the link!

Yeah, that implementation may not be a suitable one for normal use. I think a good version of this would probably have to look something like: freeze each object/array as we create it, and count on its property values having been respectively frozen when they were created, without trying to iterate through them to double-check that.

Then the one potentially tricky bit is finding a way to spell that that doesn't make all our reducers (and cached selectors) annoyingly verbose.